maintenance/shaka-player.md

```diff
@@ -0,0 +1,205 @@
+rebase of `git`.  When you run the tool, your default editor opens with a list
+of commits you may want to cherry-pick.  Any commits left in the file when you
+quit the editor are cherry-picked using `git`.
+
+If a commit fails to cherry-pick, the maintainer is given an error message from
+git and must either fix the conflict or instruct git to skip the commit.
+
+Cherry-picks can be done 1 or more times between bug-fix releases.  The release
+is finally made by merging the release PR for the branch.
+
+For example, after installing the tool:
+
+```sh
+# Go to the Shaka Player repo.
+cd /path/to/shaka-player
+
+# Check out a release branch tracking the _upstream_ repo:
+git checkout -b v3.3.x upstream/v3.3.x
+# Or if you already have such a branch locally:
+git checkout v3.3.x
+
+# Inspect all commits since a particular main-branch tag.
+# If the most recent v3.3.x release was v3.3.1, the tag
+# v3.3.1-main is the same point in history for the main branch.
+interactive-cherry-pick v3.3.1-main
+
+# If you don't use the recommended "upstream" remote name,
+# you can specify the name of your remote.  For example,
+# if you named the shaka-project/shaka-player remote "google",
+# you could run this to cherry-pick from that specific remote:
+interactive-cherry-pick v3.3.1-main google
+```
+
+Pre-work:
+ 1. Install the interactive cherry-pick tool:
+    https://github.com/shaka-project/shaka-github-tools/tree/main/interactive-cherry-pick#installation
+ 1. Create a local branch in Shaka Player tracking an upstream release branch
+
+Cherry-picking:
+ 1. Cherry-pick the necessary changes **skipping all feature commits** (see also
+    ["What to Pick"](#what-to-pick) below)
+
+    > :information_source: **NOTE**: If you aren't sure about a commit, you can
+    > select it now and change your mind later if there are merge conflicts
+    > and/or test failures
+
+ 1. Fix any merge conflicts that come up, or choose to skip the current commit
+    with `git cherry-pick --skip`
+ 1. Run the compiler with `python build/all.py`
+ 1. Run the tests with `python build/test.py`
+ 1. If any tests fail, make the necessary changes to fix the cherry-picked
+    commit that caused the failure (either to amend the earlier commits, or as
+    a new commit on top)
+ 1. When the build and all tests pass, push the cherry-picked commits upstream
+    with `git push`
+
+
+### What to Pick
+
+ - Most bug fixes are cherry-picked to active release branches.
+ - Some bug fixes won't apply to a branch because they deal with features not
+   present in that branch.
+ - Some bug fixes won't apply to a branch because the code in the branch is too
+   different, in which case [a decision must be made](https://quotecatalog.com/quote/mitch-hedberg-i-write-jokes-f-jpXNBB1)
+   about whether porting the fix to the branch is worth the effort.  The
+   maintainer will have to consider both the severity of the issue and the
+   difficulty of fixing conflicts or reimplementation.
```

Comment:
In the case of fixing conflicts or reimplementation, does the new version of the change need to be submitted as a pull request?

Reply:
Done.  PTAL!